File: employee-web-kt/src/main/java/com/moecm/Employee.kt
package com.moecm

data class Employee(
    val id: String,
    val name: String,
    val gender: String,
    val age: Int,
    val salary: Double,
    val bonus: Double,
) {
    @Suppress("unused")
    val annualSalary get() = this.salary * 12 + this.bonus
}

File: employee-web-ui/src/lib/Employee.ts
export interface Employee {
  id: string
  name: string
  gender: string
  age: number
  salary: number
  bonus: number
  annualSalary?: number
}

export function initEmployee(): Employee {
  return { id: '', name: '', gender: '', age: 0, salary: 0, bonus: 0 }
}
